src/model_selection/crossValScore.ts

```diff
@@ -0,0 +1,219 @@
+    groups?: Scikit1D
+    scoring: (this: T, X: Tensor2D) => Scalar
+  }
+): Promise<Tensor1D>
+
+/**
+ * Evaluates a score by cross-validation. This particular overload
+ * of the function uses the default score of a supervised estimator
+ * for scoring.
+ *
+ * @param estimator A supervised estimator that has an async `fit(X,y)`
+ *                 function and a `score(X,y)` function.
+ *
+ * @param Xy       A two element array containing the features and targets
+ *                 of the cross validation dataset.
+ *
+ * @param params   Cross validation parameters. `cv` is the {@link CrossValidator}
+ *                 responsible for splitting the dataset into training in test
+ *                 data in one or more folds. `groups` is additional grouping
+ *                 information which is used by some {@link CrossValidator}
+ *                 instance to determine which groups of datapoints may not
+ *                 appear both in test and training data at the same time.
+ *
+ * @returns        The test score for each split/fold that was generated by
+ *                 the {@link CrossValidator} `cv`
+ */
+export async function crossValScore(
+  estimator: {
+    fit(X: Tensor2D, y: Tensor1D): Promise<unknown>
+    score(X: Tensor2D, y: Tensor1D): Scalar
+  },
+  Xy: [Scikit2D, Scikit1D],
+  params: {
+    cv?: CrossValidator
+    groups?: Scikit1D
+  }
+): Promise<Tensor1D>
+
+/**
+ * Evaluates a score by cross-validation. This particular overload
+ * of the function uses the default score of an unsupervised estimator
+ * for scoring.
+ *
+ * @param estimator A supervised estimator that has an async `fit(X)`
+ *                 function and a `score(X)` function.
+ *
+ * @param X        An single element array containing the features of the
+ *                 cross validation dataset.
+ *
+ * @param params   Cross validation parameters. `cv` is the {@link CrossValidator}
+ *                 responsible for splitting the dataset into training in test
+ *                 data in one or more folds. `groups` is additional grouping
+ *                 information which is used by some {@link CrossValidator}
+ *                 instance to determine which groups of datapoints may not
+ *                 appear both in test and training data at the same time.
+ *
+ * @returns        The test score for each split/fold that was generated by
+ *                 the {@link CrossValidator} `cv`
+ */
+export async function crossValScore(
+  estimator: {
+    fit(X: Tensor2D): Promise<unknown>
+    score(X: Tensor2D): Scalar
+  },
+  X: [Scikit2D],
+  params: {
+    cv?: CrossValidator
+    groups?: Scikit1D
+  }
+): Promise<Tensor1D>
+
+export async function crossValScore(
+  estimator: any,
+  [X, y]: [Scikit2D, Scikit1D?],
+  {
+    cv = new KFold(),
+    groups,
+    scoring
+  }: {
+    cv?: CrossValidator
+    groups?: Scikit1D
+    scoring?: any
+  } = {}
+): Promise<Tensor1D> {
+  if (undefined === scoring) {
```

Comment:
I always prefer `scoring === undefined` (instead of the reverse) cause it reads like a sentence in my brain. "If scoring is undefined then blah blah". This is personal preference and not a huge deal.

Reply:
I tend to do it the other way around because it prevents accidents like `if(scoring = undefined)`. I changed it.